plugins/processors/scaler/scaler.go

```diff
@@ -0,0 +1,114 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package scaler
+
+import (
+	_ "embed"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Scaler) SampleConfig() string {
+	return sampleConfig
+}
+
+type Scaling struct {
+	InMin  float64  `toml:"input_minimum"`
+	InMax  float64  `toml:"input_maximum"`
+	OutMin float64  `toml:"output_minimum"`
+	OutMax float64  `toml:"output_maximum"`
+	Fields []string `toml:"fields"`
+}
+
+type Scaler struct {
+	Scalings   []Scaling       `toml:"scaling"`
+	Log        telegraf.Logger `toml:"-"`
+	scalingMap map[filter.Filter]*Scaling
+}
+
+func (s *Scaler) Init() error {
+	s.scalingMap = make(map[filter.Filter]*Scaling)
+
+	// convert filter list to filter map for better performance
+	for i, element := range s.Scalings {
+		fieldFilter, err := filter.Compile(element.Fields)
+		if err != nil {
+			s.Log.Errorf("Could not compile filter: %v\n", err)
+			return nil
+		}
+
+		if element.InMax == element.InMin {
+			s.Log.Error("Found scaling with equal input_minimum and input_maximum. Skipping it.")
+			continue
+		}
+
+		s.scalingMap[fieldFilter] = &s.Scalings[i]
+	}
+
+	return nil
+}
+
+// scale a float according to the input and output range
+func Scale(value float64, inMin float64, inMax float64, outMin float64, outMax float64) float64 {
+	return (value-inMin)*(outMax-outMin)/(inMax-inMin) + outMin
+}
```

Comment:
Please do not export this function! I would even make this a member function of `Scaling`, what do you think?
```suggestion
func (s *Scaling) Process(value float64) float64 {
	return (value - s.InMin) * (s.OutMax - s.OutMin) / (s.InMax - s.InMin) + s.OutMin
}
```
You might even want to save some cycles by pre-computing the scaling factor as
```golang
s.factor = (s.OutMax - s.OutMin) / (s.InMax - s.InMin) 
```